src/_pytest/python.py

```diff
@@ -594,9 +594,12 @@ def isinitpath(self, path):
         return path in self.session._initialpaths
 
     def collect(self):
-        path = self.fspath.dirpath()
+        this_path = self.fspath.dirpath()
         pkg_prefix = None
-        for path in path.visit(fil=lambda x: 1, rec=self._recurse, bf=True, sort=True):
+        for path in this_path.visit(rec=self._recurse, bf=True, sort=True):
+            # we will visit our own __init__.py file, in which case we skip it
```

Comment:
This is the actual fix. I did not understand what this had to do with `pytest_ignore_collect` though.

Reply:
`visit()` calls back the `self._recurse(path)` which in turn calls the `pytest_ignore_collect`